### PR TITLE
add Mellanox-npapi-thermal-psu-design

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
add Mellanox-npapi-thermal-psu-design
for thermal,
    2 APIs of 5 are supported:
        def get_temperature(self)
        def get_high_threshold(self)
    the following 3 APIs are supported:
        def get_low_threshold(self)
        def set_high_threshold(self, temperature)
        def set_low_threshold(self, temperature)

for psu,
    4 APIs of 6 are supported:
        def get_voltage(self)
        def get_current(self)
        def get_power(self)
        def get_powergood_status(self)
    the following 2 APIs are unsupported:
        def set_status_led(self, color)
        def get_status_led(self, color)
    
for fan inside psu,
    1 API of 6 is supported:
        def get_speed(self)
    the following APIs are unsupported:
        def get_direction(self)
        def get_target_speed(self)
        def get_speed_tolerance(self)
        def set_speed(self, speed)
        def set_status_led(self, color)
**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: doc/Mellanox-npapi-thermal-psu-design [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+1. cpu_core:
+    cpu_core<n>, contains the temperature of the core, for api "get_temperature".
+    cpu_core_alarm<n>, contains the threshold when temperature exceeds some cooling action is recommended. it can be used for api "get_high_threshold".
+2. cpu_pack: the files prefixed with cpu_pack have the same meaning as those with cpu_core.
+3. sfp module:
+    temp_input_module<n>, contains the temperature of the sfp module, for api "get_temperature"
+    temp_crit_module<n>, contains the critical threshold, for api "get_high_threshold"
+4. psu module:
+    psu<n>, contains the temperature of the psu module, for api "get_temperature"
+    no support for "get_high_threshold"
```
Comment (keboliu) [inline doc/Mellanox-npapi-thermal-psu-design]: IMO there is a high_threshold for PSU temp

File: doc/Mellanox-npapi-thermal-psu-design [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+5. asic:
+    asic, contains the temperature of the asic, for api "get_temperature"
+    no support for "get_high_threshold"
```
Comment (keboliu) [inline doc/Mellanox-npapi-thermal-psu-design]: not sure whether this a high threshold, please confirm.

Comment (stephenxs) [reply doc/Mellanox-npapi-thermal-psu-design]: we can use ASIC ambient temperature and high threshold is supported.

File: doc/Mellanox-npapi-thermal-psu-design (hunk 1 above)
Comment (stephenxs) [reply doc/Mellanox-npapi-thermal-psu-design]: fixed.

**Pull request closed without merging**